internal/pkg/agent/cmd/uninstall.go

```diff
@@ -54,8 +54,9 @@ func uninstallCmd(streams *cli.IOStreams, cmd *cobra.Command) error {
 	if status == install.NotInstalled {
 		return fmt.Errorf("not installed")
 	}
-	if status == install.Installed && !paths.RunningInstalled() {
-		return fmt.Errorf("can only be uninstalled by executing the installed Elastic Agent at: %s", install.ExecutablePath(paths.Top()))
+
+	if status == install.Installed && !paths.InstallMarkerExists() {
+		return fmt.Errorf("cannot find %s file relative to running elastic agent; agent should be run from %s, or the install may be corrupt", paths.MarkerFileName, install.ExecutablePath(paths.Top()))
```

Comment:
When you say _agent should be run from %s_, you mean in this case the uninstall command right? At first it seems the agent itself should be run from that does not seem to be the information you want to convey.

I'd suggest
```suggestion
		return fmt.Errorf("cannot find %s file relative to running elastic agent; agent uninstall should be run from %s, or the install may be corrupt", paths.MarkerFileName, install.ExecutablePath(paths.Top()))
```